components/login/SignInKey.tsx

```diff
@@ -0,0 +1,110 @@
+import { Alert, Button, Description, Label } from "@components/forms";
+import { isValidGovEmail } from "@lib/validation";
+import React, { useReducer, useState } from "react";
+import emailDomainList from "../../email.domains.json";
+import axios from "axios";
+import { useTranslation } from "next-i18next";
+import ErrorListItem from "@components/forms/ErrorListItem/ErrorListItem";
+
+interface FormAction {
+  name: string;
+  value: string;
+}
+
+const formReducer = (state: Record<string, string>, action: FormAction) => {
+  return {
+    ...state,
+    [action.name]: action.value,
+  };
+};
+
+const SignInKey = (): React.ReactElement => {
+  const [formData, setFormData] = useReducer(formReducer, {});
+  const [errorState, setErrorState] = useState({ message: "" });
+
+  const { t } = useTranslation("login");
+
+  const handleChange = (
+    event: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>
+  ) => {
+    setFormData({
+      name: event.target.name,
+      value: event.target.value,
+    });
+  };
+
+  const handleLoginSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    setErrorState({ message: "" });
+    const loginEmail = formData["loginEmail"];
+    if (isValidGovEmail(loginEmail, emailDomainList.domains)) {
+      try {
+        await axios({
+          url: "/api/token/temporary",
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json ",
+            Authorization: `Bearer ${formData["signInKey"]}`,
+          },
+          data: {
+            email: loginEmail,
+          },
+          timeout: process.env.NODE_ENV === "production" ? 60000 : 0,
+        });
+      } catch (err) {
+        setErrorState({
+          message: t("loginSignInErrorMessage"),
+        });
+      }
+    } else {
+      setErrorState({
+        message: t("loginSignInErrorMessage"),
```

Comment:
Did not you want to use `loginEmailErrorMessage` here? I noticed that even if my email is valid (`clement.janin@cds-snc.ca`) it displays both error messages so it may be why it does that.

Reply:
Well, we don't actually know if your email is valid. We aren't checking if that user exists in the DB, we are only checking if both the email and token (together) exist. I don't think it's good practice to say: "Your email is correct, but that's a wrong password", giving potential hackers a piece of information.